From 889a15ff11fe472ceb18aba1253f842304dd77aa Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <b02247@freescale.com>
Date: Mon, 19 Aug 2013 14:36:13 +0800
Subject: [PATCH 28/50] ENGR00265111 Video freezed for a short while when
 audio direct output

Add CBLK_DISABLED_ON flag for direct output, when underrun happened,
AudioTrack can restart the DirectOutput thread.

cherry-picked from jb4.2.2 610cd31f22839ce88bb0aff1bce5993b98b7af55

Signed-off-by: Shengjiu Wang <b02247@freescale.com>
---
 services/audioflinger/Threads.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index a7264a9..5d100b6 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -1,6 +1,7 @@
 /*
 **
 ** Copyright 2012, The Android Open Source Project
+** Copyright (C) 2013 Freescale Semiconductor, Inc.
 **
 ** Licensed under the Apache License, Version 2.0 (the "License");
 ** you may not use this file except in compliance with the License.
@@ -3679,6 +3680,8 @@ AudioFlinger::PlaybackThread::mixer_state AudioFlinger::DirectOutputThread::prep
                     // it will then automatically call start() when data is available
                     android_atomic_or(CBLK_DISABLED, &cblk->mFlags);
                 } else if (last) {
+                    android_atomic_or(CBLK_DISABLED, &cblk->flags);
+                } else if (i == (count -1)){
                     mixerStatus = MIXER_TRACKS_ENABLED;
                 }
             }
-- 
1.8.0

